pkg/cluster/cluster.go

```diff
@@ -118,12 +117,16 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("invalid cluster spec: %v", err)
 	}
 
-	var phase spec.ClusterPhase
+	// Recover the cached cluster state from the status in the current state.
+	// We could be using it to check if already "Failed" later.
```

Comment:
then we can get rid of this unclear comment at line121. it is unclear to reader what does later mean. i have to look through the code to figure out what happen after line 97.